spec.bs

```diff
@@ -70,6 +70,66 @@ dictionary AuctionAdInterestGroup {
 };
 </xmp>
 
+To join an interest group, given an {{AuctionAdInterestGroup}} |group| and a double |durationSeconds|, perform the following steps. Then return a [=promise type|Promise=]:
+
+1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the "join-ad-interest-group" [=policy-controlled feature=], then return [=a promise rejected with=] a "{{NotAllowedError}}" {{DOMException}}.
+1. Let interestGroup be a new [=interest group=].
+1. Validate the given |group| and set |interestGroup|'s fields accordingly.
+  1. Set |interestGroup|'s [=interest group/expiry=] to now plus |durationSeconds|.
+  1. If the |group|["{{AuctionAdInterestGroup/owner}}"] [=url/scheme=] is HTTPS, then set |interestGroup|'s [=interest group/owner=] to |group|["{{AuctionAdInterestGroup/owner}}"]. Otherwise, return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
+  1. Set |interestGroup|'s [=interest group/name=] to |group|["{{AuctionAdInterestGroup/name}}"].
+  1. If |group|["{{AuctionAdInterestGroup/executionMode}}"] is not null:
+    1. If it is "compatibility" or "groupByOrigin", then set |interestGroup|'s [=interest group/execution mode=] to it.
+    1. Otherwise, return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
+  1. If |group|["{{AuctionAdInterestGroup/biddingLogicUrl}}"] is not null:
+    1. If it is a valid [=URL=], then set |interestGroup|'s [=interest group/bidding url=] to it.
+    1. Otherwise, return [=a promise rejected with=] a "{{TypeError}}" {{DOMException}}.
+  1. For each of |group|["{{AuctionAdInterestGroup/biddingWasmHelperUrl}}"], |group|["{{AuctionAdInterestGroup/dailyUpdateUrl}}"], |group|["{{AuctionAdInterestGroup/trustedBiddingSignalsUrl}}"], do the same as step 4.5.

```

Comment:
You can't reliably refer to "step 4.5" inside of steps, since you quickly edit the other steps to break the reference. https://github.com/tabatkins/bikeshed/issues/796 would fix this problem.

In this case, "do the same" isn't super clear since the field names aren't exactly the same. I'd probably write the concept as:

```markdown
1. For each |group member| and |interest group field| in the following table
    <table class="data">
      <thead><tr><th>Group member</th><th>Interest group field</th></tr></thead>
      <tr><td>"{{AuctionAdInterestGroup/biddingLogicUrl}}"</td><td>[=interest group/bidding url=]</td></tr>
      ...
    </table>
    1. If |group| [=map/contains=] |group member|:
        1. Let |parsedUrl| be the result of running the [=URL parser=] on |group|[|group member|].
        1. If |parsedUrl| is an error, return [=a promise rejected with=] a {{TypeError}}.
        1. Set |interest group|'s |interest group field| to |parsedUrl|.
```
(Table styling from https://w3c.github.io/tr-design/src/README#table)